Realm/ObjectServerTests/RLMCollectionSyncTests.mm

```diff
@@ -0,0 +1,411 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMSyncTestCase.h"
+
+#if TARGET_OS_OSX
+
+#pragma mark RLMSet Sync Tests
+
+@interface RLMSetObjectServerTests : RLMSyncTestCase
+@end
+
+@implementation RLMSetObjectServerTests
+
+- (void)roundTripWithKeyPath:(NSString *)keyPath
```

Comment:
Passing a `RLMSet (^)(id)` block which returns the appropriate value rather than the property name will probably make this method a lot more readable as it's slightly more typed.

Reply:
But the object is created within the method and is extracted from a Results collection. By passing the block isn't it constraining you to create the Set outside of the roundTrip method?

Reply:
No, the roundTrip method passes the object to the block. It'd be `[self roundTripWithPropertyGetter:^(RLMSetSyncObject *o) { return o.intSet; }` rather than [self roundTripWithKeyPath:@"intSet"`. More verbose at the callsite, but basically identical within the function (`getter(obj)` rather than `obj[keyPath]`) and it lets it return a `RLMSet *` rather than `id`.

Reply:
addressed this in #7190